app/utils/sentry.ts

```diff
@@ -120,128 +136,76 @@ function captureClientErrorAsBreadcrumb(error: ClientError, isFatal: boolean) {
+        userID: user?.id ?? currentUser.id,
         email: '',
         username: '',
-        extra: {
-            locale: currentUser.locale,
-            roles: currentUser.roles,
-        },
+        locale: user?.locale ?? currentUser.locale,
+        roles: user?.roles ?? currentUser.roles,
     };
-}
+};
 
-function getExtraContext() {
-    const context = {};
-
-    // TODO: Add context based on the active database
-
-    // const currentTeam = getCurrentTeam(state);
-    // if (currentTeam) {
-    //     context.currentTeam = {
-    //         id: currentTeam.id,
-    //     };
-    // }
-
-    // const currentTeamMember = getCurrentTeamMembership(state);
-    // if (currentTeamMember) {
-    //     context.currentTeamMember = {
-    //         roles: currentTeamMember.roles,
-    //     };
-    // }
-
-    // const currentChannel = getCurrentChannel(state);
-    // if (currentChannel) {
-    //     context.currentChannel = {
-    //         id: currentChannel.id,
-    //         type: currentChannel.type,
-    //     };
-    // }
-
-    // const currentChannelMember = getMyCurrentChannelMembership(state);
-    // if (currentChannelMember) {
-    //     context.currentChannelMember = {
-    //         roles: currentChannelMember.roles,
-    //     };
-    // }
-
-    // const config = getConfig(state);
-    // if (config) {
-    //     context.config = {
-    //         BuildDate: config.BuildDate,
-    //         BuildEnterpriseReady: config.BuildEnterpriseReady,
-    //         BuildHash: config.BuildHash,
-    //         BuildHashEnterprise: config.BuildHashEnterprise,
-    //         BuildNumber: config.BuildNumber,
-    //     };
-    // }
+export const getExtraContext = async (database: Database) => {
+    const context = {
+        config: {},
+        currentChannel: {},
+        currentTeam: {},
+    };
 
-    return context;
-}
+    const config = await getConfig(database);
+    if (config) {
+        context.config = {
+            BuildDate: config.BuildDate,
+            BuildEnterpriseReady: config.BuildEnterpriseReady,
+            BuildHash: config.BuildHash,
+            BuildHashEnterprise: config.BuildHashEnterprise,
+            BuildNumber: config.BuildNumber,
+        };
+    }
+    const currentTeamId = await getCurrentTeamId(database);
+    const myTeam = await getMyTeamById(database, currentTeamId);
+    const teamRoles = myTeam?.roles;
+    context.currentTeam = {
+        TeamId: currentTeamId,
+        TeamRoles: teamRoles,
+    };
 
-function getBuildTags() {
-    let tags;
+    const channel = await getCurrentChannel(database);
+    let channelRoles;
+    if (channel) {
+        const myChannel = await getMyChannel(database, channel.id);
+        channelRoles = myChannel?.roles;
+        context.currentChannel = {
+            ChannelId: channel?.id,
+            ChannelRoles: channelRoles,
+            ChannelType: channel?.type,
+        };
+    }
```

Comment:
do we care?

this questions are to trigger a discussion/conversation

Reply:
Thanks @enahum .  On v1, these information are being collected and sent to Sentry.  Maybe to be used for deeper investigation, I'm not sure.  I tried to replicate what was in place.

If they are redundant and that from v1 experience, we can remove them.  Should the need arise in the future to bring in more `contexts` as to what channel/team the user had these problems, then we can add those back.

Reply:
Yes let's remove the team and channel context and address the other comment so we csn merge this.

Once merged ping me to give you instructions before cutting the build